src/materials/nodes/NodeMaterial.js

```diff
@@ -231,9 +231,11 @@ class NodeMaterial extends Material {
 
 			} else if ( renderer.logarithmicDepthBuffer === true ) {
 
-				const fragDepth = modelViewProjection().w.add( 1 );
-
-				depthNode = fragDepth.log2().mul( cameraLogDepth ).mul( 0.5 );
+				// Note: if you manually provide a "C" constant here for the logarithmic depth calculation, remember
+				// to also include it in the shadow depth correction in AnalyticLightNode.setupShadow()!
+				const logarithmicDepth = perspectiveDepthToLogarithmicDepth( modelViewProjection().w, cameraFar );
+				const orthographicDepth = viewZToOrthographicDepth( positionView.z, cameraNear, cameraFar );
+				depthNode = cameraIsPerspective.equal( 1 ).select( logarithmicDepth, orthographicDepth );
```

Comment:
Try using `builder.camera.isPerspectiveCamera` instead of `camera Perspective`. I think we can remove `cameraIs*` nodes.